src/main/resources/testchipip/csrc/uart.cc

```diff
@@ -0,0 +1,138 @@
+//See LICENSE for license details
+
+#include "uart.h"
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#define _XOPEN_SOURCE
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifndef _WIN32
+#include <unistd.h>
+
+// name length limit for ptys
+#define PTYNAMELEN 256
+
+/* There is no "backpressure" to the user input for sigs. only one at a time
+ * non-zero value represents unconsumed special char input.
+ *
+ * Reset to zero once consumed.
+ */
+
+// This is fine for multiple UARTs because UARTs > uart 0 will use pty, not stdio
+char specialchar = 0;
+
+void sighand(int s) {
+    switch (s) {
+        case SIGINT:
+            // ctrl-c
+            specialchar = 0x3;
+            break;
+        default:
+            specialchar = 0x0;
+    }
+}
+#endif
```

Comment:
Should we offer some (compile-time)warning that some features are not supported on WIN32?

Reply:
In the Chipyard documentation (where I assume most people will use this), we state that Windows is not recommended... so that should deter most people from using Windows. However, we can add a warning here for Windows. Something like `WARNING: The UARTAdapter instantiated is untested with Windows. Many features expected to be broken`